include/aero/grid/grid.h

```diff
@@ -1,7 +1,36 @@
 #ifndef AERO_GRID_H
 #define AERO_GRID_H
 
-/// This opaque type represents a grid implemented in C.
-typedef struct aero_grid_t aero_grid_t;
+#include <aero/array/array.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// This type represents a computional grid consisting of a set of contiguous
+/// segments separated by interfaces. The grid stores the locations of the
+/// interfaces and the midpoints between them.
+typedef struct aero_grid_t {
+  /// array storing interface coordinates (in ascending order)
+  aero_array_t *interfaces;
+  /// array storing coordinates of midpoints between interfaces (in ascending
+  /// order)
+  aero_array_t *midpoints;
+  /// lower bound (minimum interface coordinate), provided for convenience
+  aero_real_t lower_bound;
+  /// upper bound (maximum interface coordinate), provided for convenience
+  aero_real_t upper_bound;
```

Comment:
I usually use the underscores to denote private members only, since I think of public members as fields in a struct. It's all a matter of taste, but I am sensitive to the Foolish brand of consistency, so I err on the side of expressiveness and intention, which usually fall afoul of these kinds of rules. If Google's style guide advocates for underscores for public and private members without distinguishing between them, that to me is a weakness, since conveying functionality is more important than reminding the programmer of what kind of variable something is (though I guess `auto`-happy people have made this latter issue worse in recent times).

All that said, I'm happy to fall in line if we want consistency. Just giving my reasoning. Personally, `_` to me says "this doesn't belong to you."

Reply:
I'm ok with `_` just for private data members. Really, I'm just interested in consistency. I just checked and the Google style doesn't distinguish (although the example they give is for private data members), but I have no problems using our own hybrid style.